.dockerignore

```diff
@@ -1,6 +1,13 @@
-target
-Dockerfile
-.dockerignore
-.git
-.github
-.gitignore
+# Exclude everything:
+#
+*
+
+# Now un-exclude required files and folders:
```

Comment:
Question: What is the goal of this cache? When is it used?

If we're not caching the compiled binaries, how does the cache speed things up?

Reply:
The `.dockerignore` ignores the file that should not completely invalidate the cache (mainly the `cargo build` and `cargo test`)

If the cache is fully valid, then the previous cache is used, and the build won't happen.

Reply:
I'm still confused by this explanation, maybe because it contains 3 negatives (ignores, should not, invalidate).

Can you add a short comment to the file explaining how the cache works?

Reply:
This `ignore` file affects the caching of Docker image layers during image build - if there is a changed file in one of the build commands in the Dockerfile (each command 'ADD" or 'CMD' for example) creates a new layer, and subsequent builds of this image will re-use unchanged layers. The more files you are including in your image when uploading and building, the more things that can change on each build and invalidate cached layers that you may not intend. 

I'm willing to try this for now, but we may need to maintain this `.dockerignore` carefully

Reply:
I added a comment to contextualize the use of the `.dockerignore`, hope that helps a bit: https://github.com/ZcashFoundation/zebra/pull/4254/commits/7796690aea2685ceb127192065c6114bfc1ef0b5

Reply:
👍🏻